src/gtc/numpy/npir_codegen.py

```diff
@@ -160,9 +156,15 @@ def add_declared(self, *args):
         "{name} = Field({name}, _origin_['{name}'], ({', '.join(dimensions)}))"
     )
 
-    TemporaryDecl = FormatTemplate(
-        "{name} = Field.empty((_dI_ + {padding[0]}, _dJ_ + {padding[1]}, _dK_), {dtype}, ({', '.join(offset)}, 0))"
-    )
+    def visit_TemporaryDecl(
+        self, node: npir.TemporaryDecl, **kwargs
+    ) -> Union[str, Collection[str]]:
+        shape = [f"_dI_ + {node.padding[0]}", f"_dJ_ + {node.padding[1]}", "_dK_"] + [
+            str(dim) for dim in node.data_dims
+        ]
+        offset = [str(off) for off in node.offset] + ["0"] * (1 + len(node.data_dims))
```

Comment:
I am not familiar with numpy backend, but why do you need `+ 1` offsets?

Reply:
That's because `node.offset` contains only the `I` and `J` offsets, and temporaries in the `npir` backend are always `IJK` (+ perhaps some data dimensions).